Crisper Lines

danvk opened the pull request:
There were a few place where we weren't drawing lines on half-pixels. Here's a before/after:
![blurry](https://cloud.githubusercontent.com/assets/98301/10802900/b398772e-7d95-11e5-8b27-d891bac5be0a.png)
![crisp](https://cloud.githubusercontent.com/assets/98301/10802898/b1437528-7d95-11e5-8946-723794920904.png)

Fixes #335 

<!-- Reviewable:start -->
[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/hammerlab/pileup.js/341)
<!-- Reviewable:end -->

armish commented:
LTGM.

It is interesting to see how a 0.5 difference increases the quality of the drawing; I wish there were a way to optimize all these in an automated manner. All those 0.5s look weird in the code for a person who doesn't know what that does :confused: 
<!-- Reviewable comment -K1oA59tl-8c-_qO-1xJ:-1914625073 -->

---

Reviewed 5 of 5 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/hammerlab/pileup.js/341)
<!-- Sent from Reviewable.io -->

danvk commented:
Here's a [nice diagram][1] of why drawing straight lines on half-pixels looks crisper. I agree that it looks a little strange. There's an argument for moving the half pixel shifts into utility code, maybe methods like `canvasutils.drawHorizontalLine`?

I've rebased this PR and added screenshot diffs. You can really see exactly what's going on if you blink them in `webdiff`.

[1]: http://fortuito.us/diveintohtml5/canvas.html#pixel-madness
<!-- Reviewable comment -K1oIK49joXUXJVNnWaD:-423143374 -->

---

Review status: 5 of 12 files reviewed at latest revision, all discussions resolved.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/hammerlab/pileup.js/341)
<!-- Sent from Reviewable.io -->

armish commented:
yeah, moving that shift logic into the utility makes sense for longer term, but this looks great as it is :)
<!-- Reviewable comment -K1oJXB-MkT-BKaqLoks:-1408332705 -->

---

Reviewed 7 of 7 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/hammerlab/pileup.js/341)
<!-- Sent from Reviewable.io -->
